cmd/kubelet/app/init_windows.go

```diff
@@ -0,0 +1,49 @@
+// +build windows
+
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"os"
+
+	"k8s.io/kubernetes/pkg/windows/service"
+
+	"github.com/golang/glog"
+	"github.com/spf13/pflag"
+)
+
+func initForOS() {
+	// On Windows, this may be launching as a service or with an option to
+	// register the service.
+	stop, runAsService, err := service.InitService()
+	if err != nil {
+		glog.Fatal(err)
+	}
+
+	if stop {
+		os.Exit(0)
+	}
+
+	if runAsService {
+		glog.Infof("Running as service!")
+	}
+}
+
+func installFlagsForOS(fs *pflag.FlagSet) {
```

Comment:
I'd rather have this live in the `options` package with the other flag installers. Maybe in a new file called `k8s.io/kubernetes/cmd/kubelet/app/options/osflags_{windows, other, etc}.go`?

Reply:
e.g. `k8s.io/kubernetes/cmd/kubelet/app/options/osflags_windows.go`:
```
func AddOSFlags(fs *pflag.FlagSet) {
  service.InstallServiceFlags(fs, componentKubelet, "Kubernetes Kubelet")
}
```

Reply:
I would also look at how the other components in this PR install flags, and try to be consistent with their respective flag-installation patterns.

Reply:
Done.